plugin/pkg/admission/storage/persistentvolume/label/admission.go

```diff
@@ -172,15 +172,31 @@ func (l *persistentVolumeLabel) findVolumeLabels(volume *api.PersistentVolume) (
 	existingLabels := volume.Labels
 
 	// All cloud providers set only these two labels.
-	domain, domainOK := existingLabels[v1.LabelFailureDomainBetaZone]
-	region, regionOK := existingLabels[v1.LabelFailureDomainBetaRegion]
+	topologyLabelGA := true
+	domain, domainOK := existingLabels[v1.LabelTopologyZone]
+	region, regionOK := existingLabels[v1.LabelTopologyRegion]
+	// If they dont have GA labels we should check for failuredomain beta labels
+	// TODO: remove this once all the cloud provider change to topology labels
+	if !domainOK || !regionOK {
+		topologyLabelGA = false
+		domain, domainOK = existingLabels[v1.LabelFailureDomainBetaZone]
+		region, regionOK = existingLabels[v1.LabelFailureDomainBetaRegion]
+	}
+
 	isDynamicallyProvisioned := metav1.HasAnnotation(volume.ObjectMeta, persistentvolume.AnnDynamicallyProvisioned)
 	if isDynamicallyProvisioned && domainOK && regionOK {
 		// PV already has all the labels and we can trust the dynamic provisioning that it provided correct values.
+		if topologyLabelGA {
+			return map[string]string{
+				v1.LabelTopologyZone:   domain,
+				v1.LabelTopologyRegion: region,
+			}, nil
+		}
 		return map[string]string{
```

Comment:
is this reachable anymore? if domainOk && regionOK  then l181 won't be reached and topologyLabelGA must be true

Reply:
topologyLabelGA will be set to false when we could not found GA labels. At that time, domainOK && regionOK will also be redefined a new value in line 182&183

Reply:
ah got it